cpp/CMakeLists.txt

```diff
@@ -246,57 +250,72 @@ file(INSTALL ${LIBCXX_INCLUDE_DIR}     DESTINATION ${CMAKE_BINARY_DIR}/include/l
 ###################################################################################################
 # - find zlib -------------------------------------------------------------------------------------
 
-find_package(ZLIB REQUIRED)
+CPMFindPackage(
+  NAME ZLIB
+  GITHUB_REPOSITORY madler/zlib
+  VERSION 1.2.11
+)
 
-message(STATUS "ZLIB: ZLIB_LIBRARIES set to ${ZLIB_LIBRARIES}")
-message(STATUS "ZLIB: ZLIB_INCLUDE_DIRS set to ${ZLIB_INCLUDE_DIRS}")
+message(STATUS "ZLIB: ZLIB_LIBRARY set to ${ZLIB_LIBRARY}")
+message(STATUS "ZLIB: ZLIB_INCLUDE_DIR set to ${ZLIB_INCLUDE_DIR}")
 
-if(ZLIB_FOUND)
-    message(STATUS "ZLib found in ${ZLIB_INCLUDE_DIRS}")
+if(ZLIB_INCLUDE_DIR)
+    message(STATUS "ZLib found in ${ZLIB_INCLUDE_DIR}")
 else()
     message(FATAL_ERROR "ZLib not found, please check your settings.")
-endif(ZLIB_FOUND)
+endif(ZLIB_INCLUDE_DIR)
+
 
 ###################################################################################################
 # - find boost ------------------------------------------------------------------------------------
 
 # Don't look for a CMake configuration file
 set(Boost_NO_BOOST_CMAKE ON)
 
-find_package(
-    Boost REQUIRED MODULE
-    COMPONENTS filesystem
+CPMFindPackage(
+  NAME Boost
+  GITHUB_REPOSITORY Orphis/boost-cmake@60fb5fa
+  VERSION 1.70.0
+  FIND_PACKAGE_ARGUMENTS "COMPONENTS filesystem"
 )
 
-message(STATUS "BOOST: Boost_LIBRARIES set to ${Boost_LIBRARIES}")
-message(STATUS "BOOST: Boost_INCLUDE_DIRS set to ${Boost_INCLUDE_DIRS}")
+message(STATUS "BOOST: Boost_LIBRARY set to ${Boost_LIBRARY}")
+message(STATUS "BOOST: Boost_INCLUDE_DIR set to ${Boost_INCLUDE_DIR}")
 
-if(Boost_FOUND)
-    message(STATUS "Boost found in ${Boost_INCLUDE_DIRS}")
+if(Boost_INCLUDE_DIR)
+    message(STATUS "Boost found in ${Boost_INCLUDE_DIR}")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DBOOST_NO_CXX14_CONSTEXPR")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_NO_CXX14_CONSTEXPR")
     set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -DBOOST_NO_CXX14_CONSTEXPR")
 else()
     message(FATAL_ERROR "Boost not found, please check your settings.")
-endif(Boost_FOUND)
+endif(Boost_INCLUDE_DIR)
 
 ###################################################################################################
 # - RMM -------------------------------------------------------------------------------------------
 
-find_path(RMM_INCLUDE "rmm"
-          HINTS "$ENV{RMM_ROOT}/include")
+# FindCUDA doesn't quite set the variables it should. RMM needs this to be set to build from CPM
+set(CUDAToolkit_INCLUDE_DIR "${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}")
```

Comment:
That is what the documentation would have you believe. And it would have you believe that it "works" since 3.17 but was replaced in 3.18 with `FindCUDA` which now does it automagically if `CUDA` is in the `project` settings. In reality, where we are at which I think includes the next phase of clean up to function properly cleaning up our CUDA and compiler setup to be compatible. 3.17 and before the established variables are not working in any context. Which means that var is not accessible or passed on correctly to `RMM` through `CPM`. In 3.18 and our current setup we get `${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}` in the env correctly but the `CUDAToolkit_` prefixed vars. I think this is a reasonable stop gap until we move to redoing our compiler setup to be more in line and taking advantage of the 3.18 CUDA / compiler setup fully.

Reply:
`FindCUDA` is deprecated and not used in either RMM or cuDF so I'm really confused as to what's happening here.

If RMM is installed and found it should hit this `find_dependency(CUDAToolkit)` https://github.com/rapidsai/rmm/blob/branch-0.16/cmake/rmm-config.cmake.in#L12.

If RMM is added as opposed to found, it uses the targets from the `CUDAToolkit` module (https://github.com/rapidsai/rmm/blob/branch-0.16/CMakeLists.txt#L101-L103).

I imagine this is a variable used internally within some of these targets, but we really shouldn't need to mess with it.